src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java

```diff
@@ -584,6 +594,54 @@ public boolean closeWSConnection(final String wsManagerId) {
 		return false;
 	}
 
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Connect to MQTT broker.
+	 *
+	 * @param handler string id of the WebSocketcallback handler
+	 * @param brokerAddress address to the broker
+	 * @param port port to the broker
+	 * @param clientId the client name to use
+	 * @return an MQTTclient if successfull
+	 *
+	 * @throws InvalidParameterException if an error occures
+	 */
+	public MqttClient connectMQTTBroker(final MqttCallback handler, final String brokerAddress, final int port, final String caFile,
+							final String certFile, final String keyFile, final String password, final String clientId) throws Exception {
+		if (handler == null) {
+			throw new InvalidParameterException("handler cannot be null.");
+		}
+		if (Utilities.isEmpty(brokerAddress)) {
+			throw new InvalidParameterException("brokerAddress cannot be null or blank.");
+		}
+		if (port <= 0) {
+			throw new InvalidParameterException("illegal port range.");
+		}
+		if (Utilities.isEmpty(clientId)) {
+			throw new InvalidParameterException("clientId cannot be null or blank.");
+		}
+
+		final MemoryPersistence persistence = new MemoryPersistence();
+		final MqttClient client = new MqttClient(brokerAddress, clientId, persistence);
+		final MqttConnectOptions connOpts = new MqttConnectOptions();
+
+		if ( !Utilities.isEmpty(caFile) && !Utilities.isEmpty(certFile) && !Utilities.isEmpty(keyFile) ) {
+			try {
+				//connOpts.setSocketFactory(SslUtil.getSocketFactory(caFile, certFile, keyFile, password));
+			} catch(Exception err) {
+				System.out.println("MQTTS security exception: " + err);
+				System.exit(-1);
```

Comment:
What is the reason of system exit? I suggest just to throw `ArrowheadException` and let the developer decide what should happen.